sdk/spec_file_generator.go

```diff
@@ -0,0 +1,37 @@
+package sdk
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+type protocol string
+
+const (
+	protoTCP       protocol = "tcp"
+	protoNamedPipe protocol = "npipe"
+)
+
+func writeSpec(name, address string, proto protocol) (string, error) {
+
+	var pluginSpecDir string
+	if runtime.GOOS == "windows" {
+		pluginSpecDir = ([]string{filepath.Join(os.Getenv("programdata"), "docker", "plugins")})[0]
```

Comment:
The ACL concern is that the data-root of docker (which defaults to `%programdata%\docker`) MUST be set to administrators and local system, with no access to users. I want to be sure that there's no way in which the base directory can be created in a normal end-user workflow without that ACL being set. Make sense?

The location concern is that hard-coding the directory is incorrect. It must follow the data-root that the daemon is using. For example, take a look at https://github.com/docker/docker/blob/master/daemon/daemon.go#L553.